### PR TITLE
Remove our custom process implementation

### PR DESCRIPTION
**Pull request by neonichu**
It seems like if Foundation is good enough for Windows, it should be good enough everywhere.

### CODE REVIEW

**Comment by neonichu** [comment]
Somewhat WIP, since I think there might be more code that we can delete after getting rid of this.

**Comment by neonichu** [comment]
@swift-ci please test

File: Sources/TSCBasic/Process.swift [hunk 1]
```diff
@@ -528,7 +528,6 @@ public final class Process {
             throw Process.Error.missingExecutableProgram(program: executable)
         }
 
-    #if os(Windows)
         let process = Foundation.Process()
         _process = process
```
Comment (tomerd) [inline Sources/TSCBasic/Process.swift:532]: probably need to do the same treatment everywhere `_process` is used

**Comment by neonichu** [comment]
@swift-ci please test linux

**Comment by neonichu** [comment]
When I tested this locally, my changes weren't actually used, so this will need a bit more work.

**Pull request closed without merging**